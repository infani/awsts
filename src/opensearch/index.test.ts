import Opensearch from './index';

describe('index', () => {
    it('index', async () => {
        const client = await Opensearch.NewClient("url", "admin", "password")
        expect(client).toMatchSnapshot()
    });
});
