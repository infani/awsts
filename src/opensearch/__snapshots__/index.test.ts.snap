// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index index 1`] = `
Object {
  "client": Client {
    "connectionPool": ConnectionPool {
      "Connection": [Function],
      "_agent": null,
      "_proxy": null,
      "_sniffEnabled": false,
      "_ssl": null,
      "auth": null,
      "connections": Array [
        Object {
          "_openRequests": 0,
          "deadCount": 0,
          "headers": Object {},
          "id": "https://url:9200/",
          "resurrectTimeout": 0,
          "roles": Object {
            "data": true,
            "ingest": true,
          },
          "status": "alive",
          "url": "https://url:9200/",
        },
      ],
      "dead": Array [],
      "emit": [Function],
      "pingTimeout": 3000,
      "resurrectStrategy": 1,
      "resurrectTimeout": 60000,
      "resurrectTimeoutCutoff": 5,
      "size": 1,
    },
    "helpers": Helpers {
      "maxRetries": 3,
      Symbol(opensearch-client): [Circular],
      Symbol(meta header): undefined,
    },
    "name": "opensearch-js",
    "serializer": Serializer {
      Symbol(secure json parse options): Object {
        "constructorAction": "error",
        "protoAction": "error",
      },
    },
    "transport": Transport {
      "_isSniffing": false,
      "_nextSniff": 0,
      "_sniffEnabled": false,
      "compression": false,
      "connectionPool": ConnectionPool {
        "Connection": [Function],
        "_agent": null,
        "_proxy": null,
        "_sniffEnabled": false,
        "_ssl": null,
        "auth": null,
        "connections": Array [
          Object {
            "_openRequests": 0,
            "deadCount": 0,
            "headers": Object {},
            "id": "https://url:9200/",
            "resurrectTimeout": 0,
            "roles": Object {
              "data": true,
              "ingest": true,
            },
            "status": "alive",
            "url": "https://url:9200/",
          },
        ],
        "dead": Array [],
        "emit": [Function],
        "pingTimeout": 3000,
        "resurrectStrategy": 1,
        "resurrectTimeout": 60000,
        "resurrectTimeoutCutoff": 5,
        "size": 1,
      },
      "context": null,
      "emit": [Function],
      "generateRequestId": [Function],
      "headers": Object {
        "user-agent": "opensearch-js/2.0.0 (linux 5.13.0-1031-aws-x64; Node.js v16.15.1)",
      },
      "maxRetries": 3,
      "memoryCircuitBreaker": undefined,
      "name": "opensearch-js",
      "nodeFilter": [Function],
      "nodeSelector": [Function],
      "opaqueIdPrefix": null,
      "requestTimeout": 30000,
      "serializer": Serializer {
        Symbol(secure json parse options): Object {
          "constructorAction": "error",
          "protoAction": "error",
        },
      },
      "sniffEndpoint": "_nodes/_all/http",
      "sniffInterval": false,
      "sniffOnConnectionFault": false,
      "suggestCompression": false,
      Symbol(compatible check): 0,
      Symbol(api versioning): false,
    },
    Symbol(configuration error): [Function],
    Symbol(Cat): null,
    Symbol(Cluster): null,
    Symbol(DanglingIndices): null,
    Symbol(Features): null,
    Symbol(Indices): null,
    Symbol(Ingest): null,
    Symbol(Nodes): null,
    Symbol(Shutdown): null,
    Symbol(Snapshot): null,
    Symbol(Tasks): null,
    Symbol(opensearchjs-initial-options): Object {
      "Connection": [Function],
      "ConnectionPool": [Function],
      "Serializer": [Function],
      "Transport": [Function],
      "agent": null,
      "auth": null,
      "compression": false,
      "context": null,
      "disablePrototypePoisoningProtection": false,
      "enableMetaHeader": true,
      "generateRequestId": null,
      "headers": Object {},
      "maxRetries": 3,
      "name": "opensearch-js",
      "node": "https://admin@password@url:9200",
      "nodeFilter": null,
      "nodeSelector": "round-robin",
      "opaqueIdPrefix": null,
      "pingTimeout": 3000,
      "proxy": null,
      "requestTimeout": 30000,
      "resurrectStrategy": "ping",
      "sniffEndpoint": "_nodes/_all/http",
      "sniffInterval": false,
      "sniffOnConnectionFault": false,
      "sniffOnStart": false,
      "ssl": null,
      "suggestCompression": false,
    },
    Symbol(opensearchjs-extensions): Array [],
    Symbol(opensearchjs-event-emitter): EventEmitter {
      "_events": Object {},
      "_eventsCount": 0,
      "_maxListeners": undefined,
      Symbol(kCapture): false,
    },
  },
}
`;
